backend/pong/ws/tournament/async_db_serivice.py

```
@@ -177,23 +270,59 @@
-        if isinstance(e.detail, list):
-            return UpdateParticipationResult.error(
-                {"ValidationError": e.detail}
-            )
-        return UpdateParticipationResult.error(e.detail)
+        return UpdateParticipationResult.error(handle_validation_error(e))
 
     except DatabaseError as e:
-        logger.error(f"DatabaseError: {e}")
         return UpdateParticipationResult.error(
-            {"DatabaseError": f"Failed to create account. Details: {str(e)}."}
+            handle_database_error(e, "Failed to update participation.")
```

Ruff (0.8.2)

Undefined name `handle_database_error`

(F821)